// @clerk/themes/dist/clerk-js/src/ui/foundations/defaultFoundations.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultInternalThemeFoundations = void 0;
const borders_1 = require("./borders");
const colors_1 = require("./colors");
const opacity_1 = require("./opacity");
const shadows_1 = require("./shadows");
const sizes_1 = require("./sizes");
const transitions_1 = require("./transitions");
const typography_1 = require("./typography");
const zIndices_1 = require("./zIndices");
const defaultInternalThemeFoundations = Object.freeze({
    colors: colors_1.colors,
    fonts: typography_1.fonts,
    fontStyles: typography_1.fontStyles,
    fontSizes: typography_1.fontSizes,
    fontWeights: typography_1.fontWeights,
    letterSpacings: typography_1.letterSpacings,
    lineHeights: typography_1.lineHeights,
    radii: sizes_1.radii,
    sizes: sizes_1.sizes,
    space: sizes_1.space,
    shadows: shadows_1.shadows,
    transitionProperty: transitions_1.transitionProperty,
    transitionTiming: transitions_1.transitionTiming,
    transitionDuration: transitions_1.transitionDuration,
    opacity: opacity_1.opacity,
    borderStyles: borders_1.borderStyles,
    borderWidths: borders_1.borderWidths,
    zIndices: zIndices_1.zIndices,
});
exports.defaultInternalThemeFoundations = defaultInternalThemeFoundations;
