// @clerk/themes/dist/clerk-js/src/ui/foundations/typography.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.fontStyles = exports.fonts = exports.lineHeights = exports.letterSpacings = exports.fontWeights = exports.fontSizes = void 0;
const fontWeights = Object.freeze({
    normal: 400,
    medium: 500,
    semibold: 600,
    bold: 700,
});
exports.fontWeights = fontWeights;
const lineHeights = Object.freeze({
    normal: 'normal',
    extraSmall: '1.33333',
    small: '1.38462',
    medium: '1.41176',
    large: '1.45455',
});
exports.lineHeights = lineHeights;
const letterSpacings = Object.freeze({
    normal: 'normal',
});
exports.letterSpacings = letterSpacings;
// We want to achieve the md size to be 13px for root font size of 16px
// This is directly related to the createFontSizeScale function in the theme
// ref: src/ui/customizables/parseVariables.ts
const fontSizes = Object.freeze({
    xs: '0.6875rem',
    sm: '0.75rem',
    md: '0.8125rem',
    lg: '1.0625rem',
    xl: '1.5rem',
});
exports.fontSizes = fontSizes;
const fontStyles = Object.freeze({
    normal: 'normal',
});
exports.fontStyles = fontStyles;
const fonts = Object.freeze({
    main: 'inherit',
    buttons: 'inherit',
});
exports.fonts = fonts;
